engine/src/test/groovy/graphql/nadel/engine/StrategyTestHelper.groovy

```diff
@@ -151,47 +128,65 @@ class StrategyTestHelper extends Specification {
                            ResultComplexityAggregator resultComplexityAggregator
     ) {
 
-        def response1ServiceResult = new ServiceExecutionResult(response1)
-        def response2ServiceResult = new ServiceExecutionResult(response2)
 
-        boolean calledService1 = false
-        ServiceExecution service1Execution = { ServiceExecutionParameters sep ->
-            println printAstCompact(sep.query)
-            assert printAstCompact(sep.query) == expectedQuery1
-            calledService1 = true
-            return completedFuture(response1ServiceResult)
-        }
-        boolean calledService2 = false
-        ServiceExecution service2Execution = { ServiceExecutionParameters sep ->
-            println printAstCompact(sep.query)
-            assert printAstCompact(sep.query) == expectedQuery2
-            calledService2 = true
-            return completedFuture(response2ServiceResult)
-        }
+        return testServices(
+                overallSchema,
+                query,
+                [
+                        new TestService(name: serviceOneName, schema: underlyingOne, topLevelFields: topLevelFields, expectedQuery: expectedQuery1, response: response1),
+                        new TestService(name: serviceTwoName, schema: underlyingTwo, topLevelFields: [], expectedQuery: expectedQuery2, response: response2)
+                ],
+                serviceExecutionHooks,
+                variables,
+                resultComplexityAggregator
+        )
+    }
+
+    Object[] testServices(GraphQLSchema overallSchema,
```

Comment:
This new helper method allows tests to use an arbitrary number of services, instead of just 1 and 2.
It also allows top level fields to belong to different services. test2Services always maps top level fields to service1.